packages/react-i18n-webpack-plugin/src/react-i18n-webpack-plugin.ts

```diff
@@ -0,0 +1,264 @@
+              );
+
+              if (!fallBackExist) {
+                compilation.errors.push(
+                  `${componentPath}\n` +
+                    `${identifierName} 's arguments was not automatically filled in because` +
+                    `fallback translation file was not found at ${fallBackFileRelativePath} \n`,
+                );
+                return;
+              }
+
+              // Add a top-level fallbackLocale import
+              const fallbackLocaleID = camelCase(this.options.fallbackLocale);
+
+              const fallbackFileExpression = ParserHelpers.requireFileAsExpression(
+                componentDir,
+                path.join(componentDir, fallBackFileRelativePath),
+              );
+              ParserHelpers.addParsedVariableToModule(
+                parser,
+                fallbackLocaleID,
+                fallbackFileExpression,
+              );
+
+              // Replace i18n call arguments
+              const componentFileName = componentPath
+                .split('/')
+                .pop()!
+                .split('.')[0];
+              const id = generateID(componentFileName);
+              const chunkName = getChunkName(id);
+              const translationFactoryName = 'translationFactory';
+
+              ParserHelpers.toConstantDependency(
+                parser,
+                i18nCallArguments({
+                  id,
+                  translationFactoryName,
+                  fallbackLocale: this.options.fallbackLocale,
+                  fallbackLocaleID,
+                  translationFiles,
+                }),
+              )(expression);
+
+              // add translation function import
+              const factoryPath = path.join(
+                componentDir,
+                TRANSLATION_DIRECTORY_NAME,
+                'translationFactory.js',
+              );
+              const factorySource = buildFactorySource(chunkName);
+              virtualModules.writeModule(factoryPath, factorySource);
+
+              const asyncTranslationFactoryExpression = ParserHelpers.requireFileAsExpression(
+                parser.state.module.context,
+                factoryPath,
+              );
+              ParserHelpers.addParsedVariableToModule(
+                parser,
+                translationFactoryName,
+                asyncTranslationFactoryExpression,
+              );
+            });
+        };
+
+        normalModuleFactory.hooks.parser
+          .for('javascript/auto')
+          .tap('HarmonyModulesPlugin', handler);
+
+        normalModuleFactory.hooks.parser
+          .for('javascript/esm')
+          .tap('HarmonyModulesPlugin', handler);
+
+        normalModuleFactory.hooks.parser
+          .for('javascript/dynamic')
+          .tap('HarmonyModulesPlugin', handler);
+      },
+    );
+  }
+}
+
+// Return a list of translationFiles name	function i18nCallExpression(
+function getTranslationFiles(parser: any): string[] {
```

Comment:
It seems odd to pass the parser here; don't you have the parent directory of the file being processed?

Reply:
I mainly need `parser.state.compilation.compiler.inputFileSystem` from parser.